src/Symfony/Bridge/Twig/Extension/RoutingExtension.php

```diff
@@ -82,9 +87,11 @@ public function getUrl($name, $parameters = array(), $schemeRelative = false)
      * - path('route', {'param1': 'value1', 'param2': 'value2'})
      * If param1 and param2 reference placeholder in the route, it would still be safe. But we don't know.
      *
-     * @param \Twig_Node $argsNode The arguments of the path/url function
+     * @param Node $argsNode The arguments of the path/url function
      *
      * @return array An array with the contexts the URL is safe
+     *
+     * To be made @final in 3.4, and the type-hint be changed to "\Twig\Node\Node" in 4.0.
      */
     public function isUrlGenerationSafe(\Twig_Node $argsNode)
```

Comment:
We can, but that means loading Node, which in turn loads Compiler. I preferred not, thus the current code.